cosmos/ext/cosmos/ext/packet/packet.c

```diff
@@ -196,23 +196,23 @@ static VALUE packet_initialize(int argc, VALUE *argv, VALUE self)
     packet_name = argv[1];
     default_endianness = symbol_BIG_ENDIAN;
     description = Qnil;
-    buffer = rb_str_new2("");
+    buffer = Qnil;
```

Comment:
Wow optimizations in the C code. Nice! Simply don't make the allocation up front ... you think this made a big difference too? Memory?

Reply:
The buffer changes didn't make as much of a difference for the use cases we've been looking at because the packets aren't that big.  If you had 50,000 large packets, it would make a significant difference.